src/PostgREST/Response.hs

```diff
@@ -153,9 +153,13 @@ singleUpsertResponse ctxApiRequest@ApiRequest{iPreferences=Preferences{..}} resu
   RSStandard {..} -> do
     let
       response = gucResponse rsGucStatus rsGucHeaders
+      returnStatus =
+        case (rsInserted, rsUpdated) of
+          (Just i, Just u) -> if i > u then HTTP.status201 else HTTP.status200
+          _                -> HTTP.status200
```

Comment:
I updated my suggestion here: https://github.com/PostgREST/postgrest/issues/1070#issuecomment-1705901014. I think that we don't need to know exactly how many rows are updated/inserted. Anyways, this comparison only worked when we had two different settings, so it does not work when working with one. Only needs to know if rsInserted > 0.

Reply:
I saw your response there with your new approach, so you can ignore this suggestions now.